collector/datacollector/datacollector.go

```diff
@@ -20,24 +20,33 @@ package main
 import (
 	"context"
 	"flag"
+	"io/ioutil"
 	"net/http"
 	"time"
 
 	"github.com/golang/glog"
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/certificate-transparency-go/x509util"
+	"github.com/google/monologue/certgen"
 	"github.com/google/monologue/collector"
 	"github.com/google/monologue/ctlog"
 	"github.com/google/monologue/storage/print"
+	"github.com/google/trillian/crypto/keys/pem"
 )
 
 var (
 	getRootsPeriod = flag.Duration("get_roots_period", 0, "How regularly the monitor should get root certificates from the Log")
 	getSTHPeriod   = flag.Duration("get_sth_period", 0, "How regularly the monitor should get an STH from the Log")
+	addChainPeriod = flag.Duration("add_chain_period", 0, "How regularly the monitor should submit a (pre-)certificate to the Log")
 	// TODO(katjoyce): Change to read from log_list.json or all_logs_list.json to get Log details.
 	// TODO(katjoyce): Add ability to run against multiple Logs.
 	logURL    = flag.String("log_url", "", "The URL of the Log to monitor, e.g. https://ct.googleapis.com/pilot/")
 	logName   = flag.String("log_name", "", "A short, snappy, canonical name for the Log to monitor, e.g. google_pilot")
 	b64PubKey = flag.String("public_key", "", "The base64-encoded public key of the Log to monitor")
 	mmd       = flag.Duration("mmd", 24*time.Hour, "The Maximum Merge Delay for the Log")
+
+	signingCertFile = flag.String("signing_cert", "", "Path to the certificate containing the public key that corresponds to the signing key. Only needed if add_chain_period is not 0")
+	signingKeyFile  = flag.String("signing_key", "", "Path to the private key for signing certificates to submit to the Log. Only needed if add_chain_period is not 0")
```

Comment:
Couldn't those be combined? Doesn't the "normal" format for a private key file contain the public key? Or is WebPKI living up to its usual levels of disappointment?

Reply:
signingCert is the whole issuing certificate that contains the public key, not just the public key - it's for use when creating a certificate chain to send to the Log.  The private key def doesn't contain a certificate for the corresponding public key, as the cert is signed and issued after the key pair has been created.